app/main.py

```diff
@@ -1 +1,48 @@
-#  write your code here
+class Animal:
+    def __init__(
+            self,
+            name: str,
+            appetite: int,
+            is_hungry: bool = True
+    ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+
+            return self.appetite
+
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, 3, is_hungry)
+
+    def catch_mouse(self) -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, 7, is_hungry)
+
+    def bring_slippers(self) -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list[Animal]) -> int:
+    points_sum = 0
+    for animal in animals:
+        if animal.is_hungry:
+            points_sum += animal.appetite
+            animal.feed()
+
+    return points_sum
```

Comment:
Write this method in one line using `sum`. Pass generator expression there, not list comprehension.

Reply:
> Write this method in one line using `sum`. Pass generator expression there, not list comprehension.

what does generator expression have to do with it?

Reply:
You should pass a generator expression inside of `sum` to obtain the result of the `feed` operation for animals.

Reply:
> You should pass a generator expression inside of `sum` to obtain the result of the `feed` operation for animals.

I misunderstood you, thought I needed somehow to yield appetite points in the sum function😅